Tests/Unit/LoraKeysManagerFacade/ConcentratorCredentialTests.cs

```diff
@@ -146,6 +146,35 @@ public async Task RunFetchConcentratorCredentials_Returns_BadRequest_ForMissingQ
             Assert.IsType<BadRequestObjectResult>(result);
         }
 
+        [Fact]
+        public async Task RunFetchConcentratorCredentials_Returns_InternalServerError_ForMissingCupsProperty()
+        {
+            var blobBytes = Encoding.UTF8.GetBytes(RawStringContent);
+            using var blobStream = new MemoryStream(blobBytes);
+            SetupBlobMock(blobStream);
+
+            // http request
+            var httpRequest = new Mock<HttpRequest>();
```

Comment:
Consider reusing the existing logic in the test for setting up the request and the twin instead of copying the existing `arrange` part of other tests.